crates/sncast/tests/docs_snippets/validation.rs

```diff
@@ -0,0 +1,75 @@
+use docs::validation::{
+    extract_matches_from_directory, extract_matches_from_file, get_parent_dir,
+    snippet_to_command_args,
+};
+use regex::Regex;
+use tempfile::tempdir;
+
+use crate::helpers::runner::runner;
+
+#[test]
+fn test_docs_snippets() {
+    let tempdir: tempfile::TempDir = tempdir().expect("Unable to create a temporary directory");
+
+    let root_dir_path = get_parent_dir(2);
+    let docs_dir_path = root_dir_path.join("docs/src");
+    let sncast_readme_path = root_dir_path.join("crates/sncast/README.md");
+
+    let re = Regex::new(r"(?ms)```shell\n\$ (sncast .+?)\n```").expect("Invalid regex pattern");
+    let extension = Some("md");
+    let docs_snippets = extract_matches_from_directory(&docs_dir_path, &re, extension)
+        .expect("Failed to extract sncast command snippets");
+
+    let readme_snippets = extract_matches_from_file(&sncast_readme_path, &re)
+        .expect("Failed to extract sncast command snippets");
+
+    let snippets = docs_snippets
+        .into_iter()
+        .chain(readme_snippets)
+        .collect::<Vec<String>>();
+
+    let skipped_args = [
+        // snippet "$ sncast <subcommand>"
+        vec!["<subcommand>"],
```

Comment:
Why do we need that?

Reply:
It's because in docs we have such example
https://foundry-rs.github.io/starknet-foundry/starknet/index.html?highlight=subcommand#how-to-use-sncast